src/components/Input/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useRef, useEffect} from 'react';
+import type {NumberInputProps} from './NumberInput.types';
+import {BaseInput} from '../BaseInput';
+import {onArrowIncrementation} from '~/hooks';
+import {filterInput} from '~/utils/inputFilter';
+
+const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(({
+    allowNegative = true,
```

Comment:
Finally, I changed my mind; I think it's better to set `allowNegative` to `false` by default, as I believe it is the most common usage of a number input.